### DISCUSSION TITLE
Metafield List References returns null

### DISCUSSION THREAD

**Discussion by joshistoast** [Shopify/storefront-api-feedback#138] [Help] [answered]
The documentation and community forums are quite clear on how you can use references to return data from a metafield reference, but list references seem to be a different story.

Per the documentation and as I understand it, this should be all that's required to get file information from a list of file references (`list.file_reference`).
```gql
collection {
  title
  description
  gridMedia: metafield (namespace: "custom", key: "grid_media") {
    id
    value
    type
    key
    reference {
      ... on MediaImage {
        image {
          originalSrc
        }
      }
    }
  }
}
```
But alas what's returned by `reference` is `null`. This made me think I was possibly using the wrong connection, so I tried `GenericFile` and also returned null.

I've also tried using `references` instead of `reference` but I'm only met with errors. But it's clearly defined in the documentation as doing exactly what I want.

Is this a bug or am I missing something?

**Comment by joshistoast** [comment, marked as answer]
UPDATE:
I must've done something wrong, because all of a sudden it works

**Reply by SurenOO7** [reply]
How you solved this issue. I am running to the same problem.
Here is the query:
```graphQL
{
  collections(first:1) {
    nodes {
      spread: metafield(namespace: "hero", key: "spread") {
        id
        namespace
        key
        value
        type
        reference {
          ... on MediaImage {
            image {
              url
            }
          }
        }
      }
    }
  }
}
```

And this is the output: 

```JSON
{
  "data": {
    "collections": {
      "nodes": [
        {
          "spread": {
            "id": "gid://shopify/Metafield/25492367737145",
            "namespace": "hero",
            "key": "spread",
            "value": "[\"gid://shopify/MediaImage/32058750632249\",\"gid://shopify/MediaImage/32058750402873\",\"gid://shopify/MediaImage/32083388137785\",\"gid://shopify/MediaImage/32083388170553\"]",
            "type": "list.file_reference",
            "reference": null,
          }
        }
      ]
    }
  }
}
```

I have retrieved this example from `Hydrogen demo store` and changed meta field to accept multiple files (Images and Videos) instead of a single one.

It will be very helpful if you share your experience of obtaining a `reference` object.

> UPDATE

I have solved this issue by updating  `storefrontApiVersion` in `hydrogen.config` file from `2022-07` to the latest `2022-10` and instead of `reference` I used `references`.

```graphQL
{
  collections(first:1) {
    nodes {
      spread: metafield(namespace: "hero", key: "spread") {
      id
      namespace
      key
      value
      type
        references (first: 10){
          nodes {
            ... on MediaImage {
              image {
              url
              }
            }
          }
        }
      }
    }
  }
}
```

**Reply by anna-harshyna-axon** [reply]
@SurenOO7 Hi, I had the same issue, updating `storefrontApiVersion` to `2022-10` fixed it. Thanks a lot!

**Reply by kristofferso** [reply]
Hey! Are these queries still working for you? I'm using Storefront API `2021-01` but getting the same problem on `2022-10`. My metafield references are returning `[]` but `value` for the same metafield is returning data. Metafields that are not a list of references are working fine.
